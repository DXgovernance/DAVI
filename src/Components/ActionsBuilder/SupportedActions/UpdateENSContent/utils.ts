import contenthash from 'content-hash';
import { MAINNET_ID, LOCALHOST_ID, GOERLI_ID } from 'utils';
import { utils } from 'ethers';

const DEFAULT_NAMEHASH =
  '0x0000000000000000000000000000000000000000000000000000000000000000';

export const convertToNameHash = (name: string) => {
  if (!name) {
    return DEFAULT_NAMEHASH;
  }
  return utils.namehash(name);
};

export const convertToContentHash = (ipfsHash: string) => {
  if (!ipfsHash) {
    return '';
  }
  return contenthash.fromIpfs(ipfsHash);
};

export const convertToIpfsHash = (contentHash: string) => {
  if (!contentHash) return null;
  return contenthash.decode(contentHash);
};

export const getIpfsUrl = (ipfsHash: string) => {
  const ipfsRoot = 'ipfs://';
  return `${ipfsRoot}${ipfsHash}`;
};

export const isValidChainId = (chainId: number) => {
  if (!chainId || chainId === LOCALHOST_ID) {
    return MAINNET_ID;
  } else {
    return chainId;
  }
};

export const isAvailableOnENS = (chainId: number) => {
  const validChainId = isValidChainId(chainId);
  const ensNetworks = [MAINNET_ID, GOERLI_ID];
  return ensNetworks.includes(validChainId);
};
