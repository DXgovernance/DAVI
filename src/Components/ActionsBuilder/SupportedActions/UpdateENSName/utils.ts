import namehash from 'eth-ens-namehash';
import contenthash from 'content-hash';
import { useState, useEffect } from 'react';

export const convertToNameHash = (name: string) => namehash.hash(name);

export const convertToContentHash = (ipfsHash: string) =>
  contenthash.fromIpfs(ipfsHash);

export const convertToIPFSHash = (contentHash: string) => {
  if (!contentHash) return null;
  return contenthash.decode(contentHash);
};

export const useDebounce = (value: string, delay: number) => {
  const [debouncedValue, setDebouncedValue] = useState(value);

  useEffect(() => {
    const handler = setTimeout(() => {
      setDebouncedValue(value);
    }, delay);

    return () => {
      clearTimeout(handler);
    };
  }, [value, delay]);

  return debouncedValue;
};
