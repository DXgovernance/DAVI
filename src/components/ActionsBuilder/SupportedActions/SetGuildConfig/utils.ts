import { FIELDS } from './constants';
import { BigNumber } from 'ethers';
import { SetGuildConfigFields, ControlField } from './types';

export const bn = (n: string | number | BigNumber) => BigNumber.from(n);

type UpdatedValuesReturn = {
  [key in ControlField]: {
    newValue: BigNumber;
    currentValue: BigNumber;
  };
};
export const getUpdatedValues = (
  current: SetGuildConfigFields,
  modifyed: SetGuildConfigFields
): UpdatedValuesReturn | {} => {
  if (!current || !modifyed) return {};
  return Object.keys(current).reduce((acc, key) => {
    if (!FIELDS.map(f => f.name).includes(key)) return acc;
    const currentValue = bn(current[key] || 0);
    const newValue = bn(modifyed[key] || 0);

    if (!currentValue.eq(newValue)) {
      return {
        ...acc,
        [key]: { newValue, currentValue },
      };
    }
    return acc;
  }, {});
};

export const pickUpdatedOrDefaultValue = (
  current: BigNumber,
  modifyed: BigNumber
): BigNumber => {
  if (BigNumber.isBigNumber(current)) {
    if (BigNumber.isBigNumber(modifyed) && !bn(modifyed).eq(current)) {
      return bn(modifyed);
    }
    return bn(current);
  } else if (BigNumber.isBigNumber(modifyed)) {
    return bn(modifyed);
  }
  return bn(0);
};
