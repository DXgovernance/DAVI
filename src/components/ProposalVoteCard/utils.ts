import { ConfirmVoteProposalProps, VoteOnProposalProps } from './types';

export const voteOnProposal = ({
  votingPowerAtProposalSnapshot,
  votingPowerAtProposalCurrentSnapshot,
}: VoteOnProposalProps) => {
  const hasNoVotingPower =
    votingPowerAtProposalSnapshot &&
    Number(votingPowerAtProposalSnapshot?.toString()) <= 0;

  const hasVotingPowerAtCurrentSnapshot =
    votingPowerAtProposalCurrentSnapshot &&
    Number(votingPowerAtProposalCurrentSnapshot?.toString()) > 0;
  return {
    hasNoVotingPower,
    hasVotingPowerAtCurrentSnapshot,
  };
};

export const confirmVoteProposal = ({
  proposal,
  contract,
  selectedOption,
  userVotingPower,
  createTransaction,
  cb,
}: ConfirmVoteProposalProps) => {
  createTransaction(
    `Vote on proposal ${proposal?.title}`,
    async () => contract.setVote(proposal?.id, selectedOption, userVotingPower),
    true,
    cb
  );
};

export const getOptionLabel = ({ metadata, optionKey, t }) => {
  const metadataLabel = metadata?.voteOptions?.[optionKey];
  return metadataLabel
    ? metadataLabel
    : Number(optionKey) === 0
    ? t('against', { defaultValue: 'Against' })
    : t('option', { optionKey });
};
