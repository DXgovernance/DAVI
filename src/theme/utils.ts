import { css } from 'styled-components';

export const SUPPORTED_THEMES = {
  DARK: 'DARK',
  LIGHT: 'LIGHT',
};

const MEDIA_WIDTHS = {
  upToSmall: 600,
  upToMedium: 960,
  upToLarge: 1280,
};

export const mediaWidthTemplates = Object.keys(MEDIA_WIDTHS).reduce(
  (accumulator, size) => {
    accumulator[size] = (...args: any[]) => css`
      @media (max-width: ${MEDIA_WIDTHS[size]}px) {
        ${
          // @ts-ignore
          css(...args)
        }
      }
    `;
    return accumulator;
  },
  {}
);
