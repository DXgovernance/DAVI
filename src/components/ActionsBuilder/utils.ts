import { DecodedAction } from './types';
import { Permission } from './types';
import Web3 from 'web3';
import { NULL_SIGNATURE } from 'utils';
import { SupportedAction } from './types';

const web3 = new Web3();

export const getPermissionArgs = (
  decodedActions: DecodedAction[]
): Permission[] => {
  const permissions = decodedActions.map(decodedAction => {
    // Native transfer of ETH does not have a function
    if (
      decodedAction?.decodedCall?.callType ===
        SupportedAction.NATIVE_TRANSFER ||
      decodedAction?.decodedCall?.callType === SupportedAction.RAW_TRANSACTION
    ) {
      return {
        from: decodedAction?.decodedCall?.from,
        to: decodedAction?.decodedCall?.to,
        callType: decodedAction?.decodedCall?.callType,
        functionSignature: NULL_SIGNATURE,
      };
    }
    const functionInputs = decodedAction?.decodedCall?.function?.inputs?.map(
      input => input.type
    );
    const functionName = decodedAction?.decodedCall?.function?.name;
    const functionSignature = web3.eth.abi.encodeFunctionSignature(
      `${functionName}(${functionInputs.join(',')})`
    );
    return {
      from: decodedAction?.decodedCall?.from,
      to: decodedAction?.decodedCall?.to,
      callType: decodedAction?.decodedCall?.callType,
      functionSignature,
    };
  });
  return permissions;
};
