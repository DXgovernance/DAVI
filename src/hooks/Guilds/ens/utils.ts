// import { isSupportedChainId } from 'components/ActionsBuilder/SupportedActions/UpdateENSContent/utils';
import { MAINNET_ID } from 'utils';

export const isAvailableOnENS = (chainId: number) => {
  const ensNetworks = [MAINNET_ID];
  return ensNetworks.includes(chainId);
};
