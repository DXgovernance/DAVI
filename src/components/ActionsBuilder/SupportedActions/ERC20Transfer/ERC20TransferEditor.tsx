import { ActionEditorProps } from '..';
import { Button } from 'components/primitives/Button';
import { Controller, useForm } from 'react-hook-form';
import { Avatar } from 'components/Avatar';
import { TokenPicker } from 'components/TokenPicker';
import { Input } from 'components/primitives/Forms/Input';
import { TokenAmountInput } from 'components/primitives/Forms/TokenAmountInput';
import { Box } from 'components/primitives/Layout/Box';
import { BigNumber, utils } from 'ethers';
import {
  TokenInfoWithType,
  TokenType,
  useTokenList,
} from 'hooks/Guilds/tokens/useTokenList';
import { useMemo, useState } from 'react';
import { FiChevronDown } from 'react-icons/fi';
import styled from 'styled-components';
import { resolveUri } from 'utils/url';
import {
  Control,
  ControlLabel,
  ControlRow,
} from 'components/primitives/Forms/Control';
import { useTranslation } from 'react-i18next';
import { useNetwork } from 'wagmi';
import validateERC20Transfer from './validateERC20Transfer';
import { ErrorLabel } from 'components/primitives/Forms/ErrorLabel';
import { useTypedParams } from 'Modules/Guilds/Hooks/useTypedParams';
import { SupportedAction } from 'components/ActionsBuilder/types';
import ERC20 from 'contracts/ERC20.json';
import { AddressInput } from 'components/primitives/Forms/AddressInput';

const Error = styled(ErrorLabel)`
  margin-top: 0.5rem;
`;

const Spacer = styled(Box)`
  margin-right: 1rem;
`;

interface TransferValues {
  token: TokenInfoWithType;
  amount: BigNumber;
  recipientAddress: string;
}

const ERC20TransferEditor: React.FC<ActionEditorProps> = ({
  decodedCall,
  onSubmit,
}) => {
  const { t } = useTranslation();

  const { guildId } = useTypedParams();

  const { chain } = useNetwork();
  const { tokens } = useTokenList(chain?.id, true);

  const parsedData = useMemo(() => {
    if (!decodedCall) return null;

    if (decodedCall.callType === SupportedAction.ERC20_TRANSFER) {
      const token = tokens.find(token => token.address === decodedCall.to);
      return {
        source: decodedCall.from,
        token,
        amount: decodedCall.args._value,
        recipientAddress: decodedCall.args._to,
      };
    } else if (decodedCall.callType === SupportedAction.NATIVE_TRANSFER) {
      const token = tokens.find(token => token.type === TokenType.NATIVE);
      return {
        source: decodedCall.from,
        token,
        amount: decodedCall.value,
        recipientAddress: decodedCall.to,
      };
    } else {
      return null;
    }
  }, [decodedCall, tokens]);

  const { control, handleSubmit } = useForm({
    resolver: validateERC20Transfer,
    context: { t },
    defaultValues: parsedData,
  });

  const [isTokenPickerOpen, setIsTokenPickerOpen] = useState(false);

  const submitAction = (values: TransferValues) => {
    if (values.token.type === TokenType.ERC20) {
      const ERC20Contract = new utils.Interface(ERC20.abi);

      onSubmit({
        ...decodedCall,
        callType: SupportedAction.ERC20_TRANSFER,
        to: values.token.address,
        value: BigNumber.from(0),
        function: ERC20Contract.getFunction('transfer'),
        args: {
          _value: values.amount,
          _to: values.recipientAddress,
        },
      });
    } else {
      onSubmit({
        ...decodedCall,
        callType: SupportedAction.NATIVE_TRANSFER,
        to: values.recipientAddress,
        value: values.amount,
        function: null,
        args: null,
      });
    }
  };

  return (
    <div>
      <form onSubmit={handleSubmit(submitAction, console.error)}>
        <Controller
          name="recipientAddress"
          control={control}
          render={({ field: { ref, ...field }, fieldState }) => {
            const { error } = fieldState;

            return (
              <Control>
                <ControlLabel>{t('recipient')}</ControlLabel>
                <ControlRow>
                  <AddressInput
                    {...field}
                    isInvalid={!!error}
                    name="recipient-address"
                    aria-label="recipient address input"
                    placeholder={t('ethereumAddress')}
                  />
                </ControlRow>
                {!!error && <Error>{error.message}</Error>}
              </Control>
            );
          }}
        />

        <ControlRow>
          <Controller
            name="amount"
            control={control}
            render={({ field: { ref, ...field }, fieldState }) => {
              const { error } = fieldState;

              return (
                <Control>
                  <ControlLabel>{t('amount')}</ControlLabel>
                  <ControlRow>
                    <TokenAmountInput
                      {...field}
                      decimals={field.value?.decimals}
                      isInvalid={!!error}
                    />
                  </ControlRow>

                  {!!error && <Error>{error.message}</Error>}
                </Control>
              );
            }}
          />

          <Spacer />

          <Controller
            name="token"
            control={control}
            render={({ field: { ref, ...field }, fieldState }) => {
              const { error } = fieldState;
              return (
                <>
                  <Control>
                    <ControlLabel>{t('asset')}</ControlLabel>
                    <ControlRow onClick={() => setIsTokenPickerOpen(true)}>
                      <Input
                        {...field}
                        value={field.value?.symbol}
                        placeholder={t('token')}
                        isInvalid={!!error}
                        icon={
                          <div>
                            {field.value && (
                              <Avatar
                                src={resolveUri(field.value?.logoURI)}
                                defaultSeed={field.value?.address}
                                size={18}
                              />
                            )}
                          </div>
                        }
                        iconRight={<FiChevronDown size={20} />}
                        readOnly
                      />
                    </ControlRow>
                    {!!error && <Error>{error.message}</Error>}
                  </Control>

                  <TokenPicker
                    {...field}
                    walletAddress={guildId}
                    isOpen={isTokenPickerOpen}
                    onClose={() => setIsTokenPickerOpen(false)}
                    showNativeToken={true}
                    onSelect={token => {
                      field.onChange(token);
                      setIsTokenPickerOpen(false);
                    }}
                  />
                </>
              );
            }}
          />
        </ControlRow>

        <Button
          m="1rem 0 0"
          fullWidth
          data-testid="submit-erc20transfer"
          type="submit"
        >
          {t('saveAction')}
        </Button>
      </form>
    </div>
  );
};

export default ERC20TransferEditor;
