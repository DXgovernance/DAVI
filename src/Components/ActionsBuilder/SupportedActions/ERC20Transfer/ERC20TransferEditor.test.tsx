import { BigNumber } from 'ethers';
import { render } from '../../../../utils/tests';
import ERC20TransferEditor from './ERC20TransferEditor';
import {
  erc20TransferDecodedCallMock,
  erc20TransferEmptyDecodedCallMock,
} from './fixtures';

const mockBigNumber = BigNumber.from(100000000);

jest.mock('hooks/Guilds/ether-swr/ens/useENSAvatar', () => ({
  __esModule: true,
  default: () => ({
    avatarUri: 'test',
    imageUrl: 'test',
    ensName: 'test.eth',
  }),
}));

jest.mock('hooks/Guilds/ether-swr/erc20/useERC20Info', () => ({
  useERC20Info: () => ({
    name: 'Test ERC20',
    symbol: 'TEST',
    decimals: 18,
    totalSupply: mockBigNumber,
  }),
}));

const mockChainId = 123456;
jest.mock('@web3-react/core', () => ({
  useWeb3React: () => ({
    chainId: mockChainId,
  }),
}));

jest.mock('hooks/Guilds/tokens/useTokenList', () => ({
  useTokenList: () => ({
    tokens: [],
  }),
}));

jest.mock('hooks/Guilds/ether-swr/erc20/useAllERC20Balances', () => ({
  useAllERC20Balances: () => ({
    data: [],
  }),
}));

describe('ERC20TransferEditor', () => {
  it('renders', () => {
    const { container } = render(
      <ERC20TransferEditor
        decodedCall={erc20TransferDecodedCallMock}
        updateCall={jest.fn()}
      />
    );
    expect(container).toMatchSnapshot();
  });
  it('renders with default values', () => {
    const { container } = render(
      <ERC20TransferEditor
        decodedCall={erc20TransferEmptyDecodedCallMock}
        updateCall={jest.fn()}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
