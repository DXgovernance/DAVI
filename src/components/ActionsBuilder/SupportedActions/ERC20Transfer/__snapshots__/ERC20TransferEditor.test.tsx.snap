// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ERC20TransferEditor Should match snapshot 1`] = `
.c14 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  cursor: pointer;
  font-size: 14px;
  border: 1px solid #303338;
  background-color: #295FF4;
  color: #fff;
  border-radius: 32px;
  padding: 0.5rem 0.8rem;
  margin: 0.2rem;
  width: 100%;
  margin: 1rem 0 0;
}

.c14:disabled {
  color: initial;
  opacity: 0.4;
  cursor: auto;
}

.c14:hover:enabled {
  border-color: #fff;
}

.c14:active:enabled {
  border: 1px solid #303338;
}

.c6 {
  height: 1.6rem;
  width: 1.6rem;
}

.c7 {
  border-radius: 50%;
  height: 24px;
  width: 24px;
}

.c8 {
  border: none;
  background: none;
  background-color: transparent;
  box-shadow: none;
  width: 100%;
  box-sizing: border-box;
  outline: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  width: 100%;
  border: 1px solid #303338;
  border-radius: 1.5rem;
  padding: 0.5rem 0.8rem;
  background-color: transparent;
}

.c4 input {
  color: #fff;
  font-family: 'Inter',-apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-size: 14px;
  font-weight: 500;
  text-align: left;
}

.c4 input::-webkit-input-placeholder {
  color: #A1A6B0;
}

.c4 input::-moz-placeholder {
  color: #A1A6B0;
}

.c4 input:-ms-input-placeholder {
  color: #A1A6B0;
}

.c4 input::placeholder {
  color: #A1A6B0;
}

.c4:hover {
  outline: 1px solid #fff;
}

.c4:focus {
  outline: 2px solid #fff;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  width: 100%;
  border: 1px solid #303338;
  border-radius: 1.5rem;
  padding: 0.5rem 0.8rem;
  background-color: transparent;
}

.c11 input {
  color: #A1A6B0;
  font-family: 'Inter',-apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-size: 14px;
  font-weight: 500;
  text-align: left;
}

.c11 input::-webkit-input-placeholder {
  color: #A1A6B0;
}

.c11 input::-moz-placeholder {
  color: #A1A6B0;
}

.c11 input:-ms-input-placeholder {
  color: #A1A6B0;
}

.c11 input::placeholder {
  color: #A1A6B0;
}

.c11:hover {
  outline: 1px solid inherit;
}

.c11:focus {
  outline: 2px solid #fff;
}

.c5 {
  margin-right: 0.3rem;
}

.c9 {
  margin-left: 0.3rem;
}

.c0 {
  box-sizing: 'border-box';
  min-width: 0;
  margin: 0;
  padding: 0;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0.75rem 0;
  width: 100%;
}

.c2 {
  margin-bottom: 0.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  color: #A1A6B0;
  font-size: 14px;
  font-weight: 500;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c13 {
  margin-right: 1rem;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

<div>
   
  <div>
    <form>
      <div
        class="c0 c1"
      >
        <div
          class="c0 c2"
        >
          recipient
        </div>
        <div
          class="c0 c3"
        >
          <div
            class="c4"
          >
            <div
              class="c5"
            >
              <div>
                <img
                  alt="Avatar"
                  class="c6 c7"
                  data-testid="avatar"
                  size="24"
                  src="test"
                />
              </div>
            </div>
            <input
              class="c8"
              name="recipientAddress"
              placeholder="ethereumAddress"
              value="0xaF8eB8C3A5d9d900AA0B98e3Df0bcC17d3C5F698"
            />
            <div
              class="c9"
            >
              <div
                class="c0 c10"
              >
                <svg
                  fill="none"
                  height="18"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="18"
                    x2="6"
                    y1="6"
                    y2="18"
                  />
                  <line
                    x1="6"
                    x2="18"
                    y1="6"
                    y2="18"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c0 c3"
      >
        <div
          class="c0 c1"
        >
          <div
            class="c0 c2"
          >
            amount
          </div>
          <div
            class="c0 c3"
          >
            <div
              class="c11"
              disabled=""
            >
              <div
                class="c5"
              />
              <input
                autocomplete="off"
                autocorrect="off"
                class="c8"
                disabled=""
                inputmode="decimal"
                maxlength="79"
                minlength="1"
                name="amount"
                pattern="^[0-9]*[.,]?[0-9]*$"
                placeholder="0.0"
                spellcheck="false"
                type="text"
                value="5.0"
              />
              <div
                class="c9"
              >
                <div
                  aria-label="enable"
                  class="c0 c12"
                >
                  <svg
                    fill="none"
                    height="18"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="11"
                      rx="2"
                      ry="2"
                      width="18"
                      x="3"
                      y="11"
                    />
                    <path
                      d="M7 11V7a5 5 0 0 1 9.9-1"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c0 c13"
        />
        <div
          class="c0 c1"
        >
          <div
            class="c0 c2"
          >
            asset
          </div>
          <div
            class="c0 c3"
          >
            <div
              class="c4"
            >
              <div
                class="c5"
              >
                <div />
              </div>
              <input
                class="c8"
                name="token"
                placeholder="token"
                readonly=""
                value=""
              />
              <div
                class="c9"
              >
                <svg
                  fill="none"
                  height="20"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="6 9 12 15 18 9"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="c14"
        data-testid="submit-erc20transfer"
        type="submit"
      >
        saveAction
      </button>
    </form>
  </div>
   
</div>
`;

exports[`ERC20TransferEditor Should match snapshot with default values 1`] = `
.c13 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  cursor: pointer;
  font-size: 14px;
  border: 1px solid #303338;
  background-color: #295FF4;
  color: #fff;
  border-radius: 32px;
  padding: 0.5rem 0.8rem;
  margin: 0.2rem;
  width: 100%;
  margin: 1rem 0 0;
}

.c13:disabled {
  color: initial;
  opacity: 0.4;
  cursor: auto;
}

.c13:hover:enabled {
  border-color: #fff;
}

.c13:active:enabled {
  border: 1px solid #303338;
}

.c6 {
  height: 1.6rem;
  width: 1.6rem;
}

.c7 {
  border-radius: 50%;
  height: 24px;
  width: 24px;
}

.c8 {
  border: none;
  background: none;
  background-color: transparent;
  box-shadow: none;
  width: 100%;
  box-sizing: border-box;
  outline: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  width: 100%;
  border: 1px solid #303338;
  border-radius: 1.5rem;
  padding: 0.5rem 0.8rem;
  background-color: transparent;
}

.c4 input {
  color: #fff;
  font-family: 'Inter',-apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-size: 14px;
  font-weight: 500;
  text-align: left;
}

.c4 input::-webkit-input-placeholder {
  color: #A1A6B0;
}

.c4 input::-moz-placeholder {
  color: #A1A6B0;
}

.c4 input:-ms-input-placeholder {
  color: #A1A6B0;
}

.c4 input::placeholder {
  color: #A1A6B0;
}

.c4:hover {
  outline: 1px solid #fff;
}

.c4:focus {
  outline: 2px solid #fff;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  width: 100%;
  border: 1px solid #303338;
  border-radius: 1.5rem;
  padding: 0.5rem 0.8rem;
  background-color: transparent;
}

.c10 input {
  color: #A1A6B0;
  font-family: 'Inter',-apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-size: 14px;
  font-weight: 500;
  text-align: left;
}

.c10 input::-webkit-input-placeholder {
  color: #A1A6B0;
}

.c10 input::-moz-placeholder {
  color: #A1A6B0;
}

.c10 input:-ms-input-placeholder {
  color: #A1A6B0;
}

.c10 input::placeholder {
  color: #A1A6B0;
}

.c10:hover {
  outline: 1px solid inherit;
}

.c10:focus {
  outline: 2px solid #fff;
}

.c5 {
  margin-right: 0.3rem;
}

.c9 {
  margin-left: 0.3rem;
}

.c0 {
  box-sizing: 'border-box';
  min-width: 0;
  margin: 0;
  padding: 0;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0.75rem 0;
  width: 100%;
}

.c2 {
  margin-bottom: 0.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  color: #A1A6B0;
  font-size: 14px;
  font-weight: 500;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c12 {
  margin-right: 1rem;
}

<div>
   
  <div>
    <form>
      <div
        class="c0 c1"
      >
        <div
          class="c0 c2"
        >
          recipient
        </div>
        <div
          class="c0 c3"
        >
          <div
            class="c4"
          >
            <div
              class="c5"
            >
              <div>
                <img
                  alt="Avatar"
                  class="c6 c7"
                  data-testid="avatar"
                  size="24"
                  src="test"
                />
              </div>
            </div>
            <input
              class="c8"
              name="recipientAddress"
              placeholder="ethereumAddress"
              value=""
            />
            <div
              class="c9"
            />
          </div>
        </div>
      </div>
      <div
        class="c0 c3"
      >
        <div
          class="c0 c1"
        >
          <div
            class="c0 c2"
          >
            amount
          </div>
          <div
            class="c0 c3"
          >
            <div
              class="c10"
              disabled=""
            >
              <div
                class="c5"
              />
              <input
                autocomplete="off"
                autocorrect="off"
                class="c8"
                disabled=""
                inputmode="decimal"
                maxlength="79"
                minlength="1"
                name="amount"
                pattern="^[0-9]*[.,]?[0-9]*$"
                placeholder="0.0"
                spellcheck="false"
                type="text"
                value="0.0"
              />
              <div
                class="c9"
              >
                <div
                  aria-label="enable"
                  class="c0 c11"
                >
                  <svg
                    fill="none"
                    height="18"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="11"
                      rx="2"
                      ry="2"
                      width="18"
                      x="3"
                      y="11"
                    />
                    <path
                      d="M7 11V7a5 5 0 0 1 9.9-1"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c0 c12"
        />
        <div
          class="c0 c1"
        >
          <div
            class="c0 c2"
          >
            asset
          </div>
          <div
            class="c0 c3"
          >
            <div
              class="c4"
            >
              <div
                class="c5"
              >
                <div />
              </div>
              <input
                class="c8"
                name="token"
                placeholder="token"
                readonly=""
                value=""
              />
              <div
                class="c9"
              >
                <svg
                  fill="none"
                  height="20"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="6 9 12 15 18 9"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="c13"
        data-testid="submit-erc20transfer"
        type="submit"
      >
        saveAction
      </button>
    </form>
  </div>
   
</div>
`;
